types/lodash/ts3.1/lodash-tests.ts

```diff
@@ -3525,6 +3525,15 @@ fp.now(); // $ExpectType number
     const memoizedFn = _.memoize(memoizeFn);
     memoizedFn.cache = new WeakMap();
     memoizedFn.cache = new Map();
+
+    {
+        type MemoizedResultFn = (a1: string, a2: number) => boolean;
+        let beforeMemoize: MemoizedResultFn & {otherKey: number};
+        let result: MemoizedResultFn;
+
+        beforeMemoize = _.assign(memoizeFn, {otherKey: 1});
+        result = _.memoize(beforeMemoize);
```

Comment:
Use $ExpectType to get stricter type checking here. I think this test will currently pass even without your change.

E.g.

`_.memoize(...); // $ExpectType (a1: string, ...`

Note that ExpectType is kind of dumb (it just does a string comparison rather than actually checking if the 2 types are equivalent) so you can't use your MemoizedResultFn type. Unless they improved it recently.